kaggle/api/kaggle_api_extended.py

```diff
@@ -86,98 +86,248 @@ class KaggleApi(KaggleApi):
+
+        return config_data
+
+
+## Configuration
+
+    def _load_config(self, config_data):
+        '''the final step of the authenticate steps, where we load the values
+           from config_data into the Configuration object.
 
-            self.copy_config_value(self.CONFIG_NAME_PROXY, config_data)
-            self.copy_config_value(self.CONFIG_NAME_PATH, config_data)
-            self.copy_config_value(self.CONFIG_NAME_COMPETITION, config_data)
-            self.copy_config_value(self.CONFIG_NAME_USER, config_data)
+           Parameters
+           ==========
+           config_data: a dictionary with configuration values (keys) to read
+                        into self.config_values
 
-            configuration.username = config_data[self.CONFIG_NAME_USER]
-            configuration.password = config_data[self.CONFIG_NAME_KEY]
-            if self.CONFIG_NAME_PROXY in config_data:
-                configuration.proxy = config_data[self.CONFIG_NAME_PROXY]
+        '''
+        # Username and password are required.
+
+        for item in [self.CONFIG_NAME_USER, self.CONFIG_NAME_KEY]:
+            if item not in config_data:
+                print('Error: Missing %s in configuration.' % item)
+                sys.exit(1)
+
+        configuration = Configuration()
+
+        # Add to the final configuration (required)
+
+        configuration.username = config_data[self.CONFIG_NAME_USER]
+        configuration.password = config_data[self.CONFIG_NAME_KEY]
+
+        # Proxy
+
+        if self.CONFIG_NAME_PROXY in config_data:
+            configuration.proxy = config_data[self.CONFIG_NAME_PROXY]
+
+        # Keep config values with class instance, and load api client!
+
+        self.config_values = config_data
+
+        try:
             self.api_client = ApiClient(configuration)
 
         except Exception as error:
+
             if 'Proxy' in type(error).__name__:
-                raise ValueError(
-                    'The specified proxy ' +
-                    config_data[self.CONFIG_NAME_PROXY] +
-                    ' is not valid, please check your proxy settings')
+                sys.exit('The specified proxy ' + 
+                         config_data[self.CONFIG_NAME_PROXY] +
+                         ' is not valid, please check your proxy settings')
             else:
-                raise ValueError(
-                    'Unauthorized: you must download an API key from '
-                    'https://www.kaggle.com/<username>/account\nThen put ' +
-                    self.config_file + ' in the folder ' + self.config_dir)
+                sys.exit('Unauthorized: you must download an API key or export '
+                         'credentials to the environment. Please see\n ' +
+                         'https://github.com/Kaggle/kaggle-api#api-credentials '
+                          + 'for instructions.')
 
-    def set_config_value(self, name, value, quiet=False):
+
+    def _authenticate_config(self, quiet=False):
```

Comment:
I would consolidate this into  `_read_config`  by saving off if we've already done the file permissions check, similar to what is done with API client version check (see `process_response`).

Reply:
The reason is because the functions `authenticate_environment` and `authenticate_config` aren't the exact functions to do an action like read/write, but the higher up functions that describe a series of actions. You could imagine adding another `authenticate_from_api` or `authenticate_oauth`, each of which might use `read_config` and `write_config` (but are most definitely different)

Reply:
The problem is that a lot more is happening here (and in the env variables method) than just authentication.  It's reading in a bunch of config variables and setting them too, which is not clear from the method names.  Additionally, the authentication isn't actually happening here - it happens inside of `_load_config`

Reply:
hmm okay, how about `read_config_environment` and `read_config_file` (not with the underscore, so it's less a "private") function?

Reply:
okay! I'll push these changes right away! I just made the change locally and I like it so much better, it just reads exactly what it does. ;)